File: scratch/download_pieces.js
const fs = require('fs');
const path = require('path');

const themes = ['cburnett', 'merida', 'alpha', 'fantasy'];
const pieces = ['wK', 'wQ', 'wR', 'wB', 'wN', 'wP', 'bK', 'bQ', 'bR', 'bB', 'bN', 'bP'];
const baseUrl = 'https://raw.githubusercontent.com/lichess-org/lila/master/public/piece/';

async function downloadPieces() {
  const promises = [];
  for (const t of themes) {
    const dir = path.join(__dirname, 'frontend', 'public', 'pieces', t);
    fs.mkdirSync(dir, { recursive: true });
    
    for (const p of pieces) {
      const url = `${baseUrl}${t}/${p}.svg`;
      const filePath = path.join(dir, `${p}.svg`);
      
      promises.push(
        fetch(url)
          .then(res => {
            if (!res.ok) throw new Error(`Status ${res.status}`);
            return res.text();
          })
          .then(text => fs.promises.writeFile(filePath, text))
          .then(() => console.log(`Downloaded ${t}/${p}.svg`))
          .catch(err => console.error(`Failed ${t}/${p}.svg:`, err.message))
      );
    }
  }
  
  await Promise.all(promises);
  console.log('All done!');
}

downloadPieces();

File: scratch/download_pieces.py
import urllib.request
import os

themes = ['cburnett', 'merida', 'alpha', 'fantasy']
pieces = ['wK', 'wQ', 'wR', 'wB', 'wN', 'wP', 'bK', 'bQ', 'bR', 'bB', 'bN', 'bP']
base_url = 'https://raw.githubusercontent.com/lichess-org/lila/master/public/piece/'

for t in themes:
    os.makedirs(f'frontend/public/pieces/{t}', exist_ok=True)
    for p in pieces:
        url = f'{base_url}{t}/{p}.svg'
        path = f'frontend/public/pieces/{t}/{p}.svg'
        try:
            urllib.request.urlretrieve(url, path)
            print(f'Downloaded {path}')
        except Exception as e:
            print(f'Error downloading {path}: {e}')
